src/main/scala/com/microsoft/azure/sqldb/spark/sql/streaming/SQLWriter.scala

```diff
@@ -0,0 +1,193 @@
+import org.apache.spark.sql.catalyst.InternalRow
+import com.microsoft.azure.sqldb.spark.connect.ConnectionUtils._
+import org.apache.spark.sql.catalyst.expressions._
+
+import scala.Array
+import scala.collection.mutable.ListBuffer
+import scala.util.Try
+
+/**
+  * The [[SQLWriter]] class is used to write data from a batch query
+  * or structured streaming query, given by a [[QueryExecution]], to Azure SQL Database or Azure SQL Data Warehouse.
+  */
+private[spark] object SQLWriter extends Logging {
+
+  var subset = false
+  var DRIVER_NAME: String = "com.microsoft.sqlserver.jdbc.SQLServerDriver"
+  var connection: Connection = _
+  val daysToMillis = 24L*60L*60L*1000L
+
+  override def toString: String = "SQLWriter"
+
+  def write(
+             sparkSession: SparkSession,
+             data: DataFrame,
+             queryExecution: QueryExecution,
+             saveMode: SaveMode,
+             parameters: Map[String, String]
+           ): Unit = {
+
+    //TODO: (Future Release) Provide the option to the user to define the columns they're writing to and/or column mapping
+    var writeConfig = Config(parameters)
+    val url = writeConfig.get[String](SqlDBConfig.URL).getOrElse(
+      throw new IllegalArgumentException("URL not found in Config")
+    )
+    val db = writeConfig.get[String](SqlDBConfig.DatabaseName).getOrElse(
+      throw new IllegalArgumentException("Database not found in Config")
+    )
+    val table = writeConfig.get[String](SqlDBConfig.DBTable).getOrElse(
+      throw new IllegalArgumentException("table not found in Config")
+    )
+    val user = writeConfig.get[String](SqlDBConfig.User).getOrElse(
+      throw new IllegalArgumentException("user not found in Config")
+    )
+    val password = writeConfig.get[String](SqlDBConfig.Password).getOrElse(
+      throw new IllegalArgumentException("password not found in Config")
+    )
+    val port = writeConfig.get[String](SqlDBConfig.PortNumber).getOrElse("1433")
+    val ignoreColumnNames = Try(writeConfig.get[String]("ignoreColumnNames").map(_.toBoolean).getOrElse(false)).getOrElse(false)
+
+    //Using regular writes
+    Class.forName(DRIVER_NAME)
+    connection = DriverManager.getConnection(createJDBCUrl(url, Some(port)) + ";database=" + db, user, password)
+
+
+    /* Getting Column information */
+    try {
+
+      val schema = queryExecution.analyzed.output
+      var schemaDatatype = new ListBuffer[DataType]()
+      var accessorsArr = new ListBuffer[(SpecializedGetters, Int) => Any]()
+      var colNames = ""
+      var values = ""
+
+      if (schema.nonEmpty) {
+        schema.foreach(col => {
+          colNames += col.name + ","
+          values += "?,"
+          schemaDatatype += col.dataType
+          accessorsArr += getAccessor(col.dataType)
+        })
+      }
+
+      var sql = ""
+
+      if(ignoreColumnNames) {
+        sql = "INSERT INTO " + table + " VALUES (" + values.substring(0, values.length - 1) + ");"
```

Comment:
What if 'table' has a space, or reserved characters in the object identifier? On a similar note, what if the table has a schema name (in SQL you can have tables in other schemas). Since we are constructing dynamic SQL these edge cases might need to be handled.